syntax = "proto3";

// The greeting service definition.
service TicketService {
  rpc GetFlightsByRoute (FlightsByRouteRequest) returns (FlightsByRouteReply){}
  rpc BuyFlightPackage (BuyFlightPackageRequest) returns (BuyFlightPackageReply){}
}

message FlightsByRouteRequest{
  string src = 1;
  string dest = 2;
}

message FlightsByRouteReply{
  string flights = 1;
}

message BuyFlightPackageRequest{
  repeated int32 flights_id = 1;
  int32 seats_amount = 2;
}

message BuyFlightPackageReply{
  bool buy_success = 1;
  string message = 2;
}